modules/cce/cluster.tf

```diff
@@ -65,13 +65,22 @@ resource "opentelekomcloud_cce_cluster_v3" "cluster" {
   }
 }
 
+locals {
+  additional_ngs = distinct(flatten([
+    for az in var.node_availability_zones : [
+      for flavor in var.node_additional_flavors : "${az}|${flavor}"
+    ]
+  ]))
+  final_ng_list = toset(concat(tolist(var.node_availability_zones), local.additional_ngs))
```

Comment:
This construct creates a list
In this example: 
`node_additional_flavors = "c4.2xlarge.2"`
`node_availability_zones = ["eu-de-01","eu-de-02","eu-de-03"]`

```
eu-de-01
eu-de-02
eu-de-03
eu-de-01|c4.2xlarge.2
eu-de-02|c4.2xlarge.2
eu-de-03|c4.2xlarge.2
```
The first three entries maintain the backwards-compatibilty.

Reply:
This can be achieved by using setproduct function of terraform.
- with var.node_flavors and var.availability_zones as sets you can:
``` HCL
locals {
  node_pools = setproduct(var.node_availability_zones, var.node_flavors)
}
resource "opentelekomcloud_cce_node_pool_v3" "cluster_node_pool" {
  count = length(local.node_pools)
  cluster_id = opentelekomcloud_cce_cluster_v3.cluster.id
  flavor = local.node_pools[count.index][1]
  name = "${var.name}-nodes-${local.node_pools[count.index][0]}-${replace(local.node_pools[count.index][1], ".", "-")}"
  initial_node_count = var.node_count
  availability_zone = local.node_pools[count.index][0]
  key_pair = opentelekomcloud_compute_keypair_v2.cluster_keypair.name
  os = var.node_os
  ...
```

or even more readable:

``` HCL
locals {
  node_pools = {for element in setproduct(var.node_availability_zones, var.node_flavors) : "${element[0]}-${replace(element[1], ".", "-")}" => {
    availability_zone = element[0]
    flavor            = element[1]
  }}
}
resource "opentelekomcloud_cce_node_pool_v3" "cluster_node_pool" {
  for_each = local.node_pools
  cluster_id = opentelekomcloud_cce_cluster_v3.cluster.id
  flavor = each.value.flavor
  name = "${var.name}-nodes-${each.key}"
  initial_node_count = var.node_count
  availability_zone = each.value.availability_zone
  key_pair = opentelekomcloud_compute_keypair_v2.cluster_keypair.name
  os = var.node_os
  ...
``` 

ps: I did not test them so there might be syntax issues here and there but the general idea should be straight forward.